xmake/rules/c++/modules/modules_support/clang.lua

```diff
@@ -150,7 +150,16 @@ function generate_dependencies(target, sourcebatch, opt)
 
             -- no support of p1689 atm
             local jsonfile = path.translate(path.join(outputdir, path.filename(sourcefile) .. ".json"))
-            common.fallback_generate_dependencies(target, jsonfile, sourcefile)
+            common.fallback_generate_dependencies(target, jsonfile, sourcefile, function(file)
+                local compinst = target:compiler("cxx")
```

Comment:
```lua
        local clang, toolname = target:tool("cxx")
        assert(toolname == "clang")
        -- ...
        os.vrunv(clang, table.join(defines, {"-E", "-x", "c++", file,  "-o", ifile}))
```

Reply:
or we can use 

```lua
    local compinst = compiler.load("cxx", {target = target})
    local compflags = table.join("-E", "-x", "c++", compinst:compflags({target = target}))
    os.vrunv(compinst:program(), table.join(compflags, "-o", ifile))
```

and remove

```lua
local defines = {}
                for _, define in ipairs(target:get("defines")) do
                    table.insert(defines, "-D" .. define)
                end
```

Reply:
yeah but when some flags that are in compflags prevent -E from working (the compiler try to search module imports, we don't want that)